test/src/tests-self-hosting-specific.js

```diff
@@ -109,52 +105,75 @@ describe('/config self-hosting integration tests', function() {
+                            mpid: 'identifyMPID', is_logged_in: false
+                        }),
+                        headers: { 'Content-Type': 'application/json' },
+                    });
+                }, 5000); // 100ms delay
+            })
+        });
 
+        fetchMockSuccess(urls.login, {
+            mpid: 'loginMPID', is_logged_in: false
+        });
 
         mParticle.init(apiKey, window.mParticle.config);
-        // call login before mParticle.identify is triggered, which happens after config returns
         mParticle.Identity.login({ userIdentities: { customerid: 'abc123' } });
+        mParticle.getInstance()._Store.isInitialized = true;
         mParticle.logEvent('Test');
+        
+        waitForCondition(() => {
+                return (
+                    mParticle.Identity.getCurrentUser()?.getMPID() === 'loginMPID'
+                );
+            })
+        .then(() => {
+                // call login before mParticle.identify is triggered, which happens after config returns
+                // mParticle.Identity.login({ userIdentities: { customerid: 'abc123' } });
+        waitForCondition(() => {
+            return (
+                mParticle.getInstance()._Store.configurationLoaded === true
+            );
+        })
+            .then(() => {
+                // config triggers, login triggers immediately before identify
+                const event1 = findBatch(fetchMock.calls(), 'Test');
+                event1.mpid.should.equal('loginMPID');
+                messages
+                    .indexOf('Parsing "login" identity response from server')
+                    .should.be.above(0);
+    
+                // when login is in flight, identify will not run, but callback still will
+                messages
+                    .indexOf('Parsing "identify" identity response from server')
+                    .should.equal(-1);
 
-        // config triggers, login triggers immediately before identify
-        setTimeout(() => {
-            const event1 = findBatch(fetchMock.calls(), 'Test');
-            event1.mpid.should.equal('loginMPID');
-            messages
-                .indexOf('Parsing "login" identity response from server')
-                .should.be.above(0);
-
-            // when login is in flight, identify will not run, but callback still will
-            messages
-                .indexOf('Parsing "identify" identity response from server')
-                .should.equal(-1);
-            const event2 = findBatch(fetchMock.calls(), 'identify callback event', false, mockServer);
-            event2.mpid.should.equal('loginMPID');
-
-            mockServer.restore();
-            // clock.restore();
-
-            localStorage.removeItem('mprtcl-v4_workspaceTokenTest');
-            window.mParticle.config.requestConfig = false;
 
-            done();
-            
-        }, 150);
+                    // TODO - COME BACK TO ME
```

Comment:
Is this comment still relevant?

Reply:
no.  removing.  this is the same test as above that has a new ticket // https://go.mparticle.com/work/SQDSDKS-6852